src/jdk.compiler/share/classes/com/sun/source/tree/PatternTree.java

```diff
@@ -26,17 +26,9 @@
 package com.sun.source.tree;
 
 /**
- * {@preview Associated with pattern matching for instanceof, a preview feature of
- *           the Java language.
- *
- *           This interface is associated with <i>pattern matching for instanceof</i>, a preview
- *           feature of the Java language. Preview features
- *           may be removed in a future release, or upgraded to permanent
- *           features of the Java language.}
- *
  * A tree node used as the base class for the different kinds of
```

Comment:
the javadoc seems to need an update

Reply:
Patterns are not really expressions - they are a new kind of trees. Basically, before we had "statements" and "expressions" (+declarations, which are a bit special, ClassTree and VariableTree are "statements", but MethodTree, ModuleTree and PackageTree subtype Tree directly, so there is no common supertype). Now we have "patterns" in addition to that.

For example, having: `o instanceof String s`, this is an expression, but `String s` is not - it is a pattern. It cannot stand on its own as an expression, it is a special part of the expression.